content/blog/2023/2023-05-18-operators-in-typescript.md

```diff
@@ -0,0 +1,177 @@
+
+~~~ts
+let x = 5;
+let y = 10;
+let z = 15;
+
+console.log(x < y && y < z); // true
+console.log(x > y || y < z); // true
+console.log(!(x > y)); // true
+~~~
+
+**AND (`&&`) Operator:** The AND `&&` operator returns true if both operands are true.
+
+**OR (`||`) Operator:** The OR `||` operator returns true if at least one of the operands is true.
+
+**NOT (`!`) Operator:** The NOT `!` operator negates a boolean value, converting true to false and vice versa. 
+
+## Bitwise Operators
+Bitwise operators in TypeScript operate on binary representations of numbers. They are useful for performing operations at the binary level, manipulating individual bits, working with flags, or optimizing certain algorithms that require low-level bit manipulations. Understanding their behaviour and how to use them correctly can be valuable when working with binary data or implementing certain advanced programming techniques.
+
+Some of the common bitwise operators include AND (`&`), OR (`|`), XOR (`^`), left shift (`<<`), right shift (`>>`), and complement (`~`). These operators are particularly useful when working with flags, binary data, or performance optimizations. Let’s take a look at them:
+
+~~~ts
+let a = 5; // 0101 in binary
+let b = 3; // 0011 in binary
+
+console.log(a & b); // 0001 (AND)
+console.log(a | b); // 0111 (OR)
+console.log(a ^ b); // 0110 (XOR)
+console.log(a << 1); // 1010 (Left Shift)
+console.log(a >> 1); // 0010 (Right Shift)
+console.log(~a); // 1010 (Complement)
+~~~
+
+AND (`&`): The AND operator, represented by the symbol `&`, performs a logical AND operation between corresponding bits of two numbers. The result is a new number where each bit is set to 1 only if both bits in the same position of the operands are 1.
+
+OR (`|`): The OR operator, represented by the symbol `|`, performs a logical OR operation between corresponding bits of two numbers. The result is a new number where each bit is set to 1 if at least one of the bits in the same position of the operands is 1.
+
+XOR (`^`): The XOR operator, represented by the symbol `^`, performs a logical exclusive OR operation between corresponding bits of two numbers. The result is a new number where each bit is set to 1 only if one of the bits in the same position of the operands is 1, but not both.
+
+Left Shift (`<<`): The left shift operator, represented by the symbol `<<`, shifts the binary bits of a number to the left by a specified number of positions. The leftmost bits are discarded, and new zeros are added on the right side. Each shift to the left doubles the value of the number.
+
+Right Shift (`>>`): The right shift operator, represented by the symbol `>>`, shifts the binary bits of a number to the right by a specified number of positions. The rightmost bits are discarded, and new bits are added on the left side. For positive numbers, each shift to the right halves the value of the number.
```

Comment:
> The rightmost bits are discarded, and new bits are added on the left side

"The rightmost bits are discarded, and new **zeros** are added on the left side"